### PR TITLE
Iox #25 add locking policy to chunk queue

### PR DESCRIPTION
**Pull request by marthtz**

### CODE REVIEW

File: iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp [hunk 1]
```diff
@@ -54,245 +54,272 @@ class ChunkQueue_testBase
...
-    auto ret = m_popper.attachConditionVariable(&condVar);
+    auto ret = this->m_popper.attachConditionVariable(&condVar);
     EXPECT_TRUE(ret);
 
-    EXPECT_THAT(m_popper.isConditionVariableAttached(), Eq(true));
+    EXPECT_THAT(this->m_popper.isConditionVariableAttached(), Eq(true));
 }
 
-TEST_P(ChunkQueue_test, DISABLED_PushAndNotifyConditionVariableSignaler)
+TYPED_TEST(ChunkQueue_test, PushAndNotifyConditionVariableSignaler)
```
Comment (mossmaurice) [inline iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp:171]: `PushAndNotifyConditionVariable`?

File: iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp [hunk 2]
```diff
@@ -54,245 +54,272 @@ class ChunkQueue_testBase
...
-    m_pusher.push(chunk);
+    auto chunk = this->allocateChunk();
+    this->m_pusher.push(chunk);
 
     EXPECT_THAT(condVarWaiter.timedWait(1_ns), Eq(true));
     EXPECT_THAT(condVarWaiter.timedWait(1_ns), Eq(false)); // shouldn't trigger a second time
 }
 
-TEST_P(ChunkQueue_test, DISABLED_AttachSecondConditionVariableSignaler)
+TYPED_TEST(ChunkQueue_test, AttachSecondConditionVariableSignaler)
```
Comment (mossmaurice) [inline iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp:186]: Same

File: iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp [hunk 3]
```diff
@@ -54,245 +54,272 @@ class ChunkQueue_testBase
...
+TYPED_TEST(ChunkQueueFiFo_test, DISABLED_Capacity)
 {
-    EXPECT_THAT(m_popper.getCurrentCapacity(), Eq(iox::MAX_RECEIVER_QUEUE_CAPACITY));
+    EXPECT_THAT(this->m_popper.getCurrentCapacity(), Eq(iox::MAX_RECEIVER_QUEUE_CAPACITY));
 }
 
-
 /// @note API currently not supported
-TEST_F(ChunkQueueFiFo_test, DISABLED_SetCapacity)
+TYPED_TEST(ChunkQueueFiFo_test, DISABLED_SetCapacity)
```
Comment (mossmaurice) [inline iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp:244]: Why is it disabled?

File: iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp [hunk 4]
```diff
@@ -54,245 +54,272 @@ class ChunkQueue_testBase
...
+    using ChunkQueueData_t = ChunkQueueData<iox::DefaultChunkQueueConfig, PolicyType>;
 
     ChunkQueueData_t m_chunkData{iox::cxx::VariantQueueTypes::FiFo_SingleProducerSingleConsumer};
     ChunkQueuePopper<ChunkQueueData_t> m_popper{&m_chunkData};
     ChunkQueuePusher<ChunkQueueData_t> m_pusher{&m_chunkData};
 };
 
 /// @note API currently not supported
-TEST_F(ChunkQueueFiFo_test, DISABLED_InitialSize)
+TYPED_TEST(ChunkQueueFiFo_test, DISABLED_InitialSize)
```
Comment (mossmaurice) [inline iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp:232]: Why disabled?

File: iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp (hunk 1 above)
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp]: Done

File: iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp (hunk 2 above)
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp]: Done

File: iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp (hunk 3 above)
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp:244]: See your comment above the test `/// @note API currently not supported`

File: iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp (hunk 4 above)
Comment (marthtz) [reply iceoryx_posh/test/moduletests/test_popo_chunk_queue.cpp:232]: See your comment above the test `/// @note API currently not supported`

**Pull request merged**